[KIP-848] Added new configs group.protocol and group.remote.assignor (#4414)
confluentinc/librdkafka: CONFIGURATION.md

session.timeout.ms                       |  C  | 1 .. 3600000    |         45000 | high       | Client group session and failure detection timeout. The consumer sends periodic heartbeats (heartbeat.interval.ms) to indicate its liveness to the broker. If no hearts are received by the broker for a group member within the session timeout, the broker will remove the consumer from the group and trigger a rebalance. The allowed range is configured with the **broker** configuration properties `group.min.session.timeout.ms` and `group.max.session.timeout.ms`. Also see `max.poll.interval.ms`. <br>*Type: integer*
heartbeat.interval.ms                    |  C  | 1 .. 3600000    |          3000 | low        | Group session keepalive heartbeat interval. <br>*Type: integer*
group.protocol.type                      |  C  |                 |      consumer | low        | Group protocol type. NOTE: Currently, the only supported group protocol type is `consumer`. <br>*Type: string*

> group.protocol                           |  C  | generic, consumer |       generic | high       | Group protocol to use. Use `generic` for the old protocol and `consumer` for the new protocol introduced in KIP-848. Availabe protocols: generic or consumer. Default is `generic` but is expected to be changed to `consumer` in future. <br>*Type: enum value*

Reviewer: ```suggestion
group.protocol                           |  C  | generic, consumer |       generic | high       | Group protocol to use. Use `generic` for the original protocol and `consumer` for the new protocol introduced in KIP-848. Available protocols: generic or consumer. Default is `generic`, but will change to `consumer` in next releases. <br>*Type: enum value*
```

Reviewer: Change to `Group protocol type for the generic protocol` in `group.protocol.type` description.

Author: I am not touching `group.protocol.type`. It is an old property with different use.

Reviewer: Only the documentation, as it's for the generic protocol, otherwise it's confusing
